### PR TITLE
#DP-98 ft: Error message

### PR DESCRIPTION
**Pull request by jeandamascyifuzo**
### PR Description
Custom Error responses

### How should this be manually tested?

**Run the commands below in your terminal**
git checkout fix-custom-err-res-dp-98
git fetch
npm install
npm run dev
open http://localhost:4000/

check Whether error messages are displayed correctly

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,9 +11,11 @@ const server = new ApolloServer({
     typeDefs,
     resolvers,
     introspection: true,
+    debug:false,
```
Comment (rukundo-kevin) [inline src/index.ts]: This works, but isn't a way we can deduct the stacktrace from the client response but still have it for debugging purposes as this make it completely unavailable.

Comment (feyton) [reply src/index.ts]: That would be the best compromise in between. We need errors for debugging our code when things break.
We can try based on the `NODE_ENV`

File: src/schema/index.ts [hunk 2]
```diff
@@ -4,11 +4,13 @@ const Schema = gql`
   type User {
     id: ID!
     role: String!
+    name: String
```
Comment (feyton) [inline src/schema/index.ts]: Why did we add a field here which is not related to the user object?

File: src/schema/index.ts [hunk 3]
```diff
@@ -4,11 +4,13 @@ const Schema = gql`
...
     email: String!
     password: String!
     profile: Profile!
   }
   input RegisterInput {
+    name: String
```
Comment (feyton) [inline src/schema/index.ts]: Same here, this field is not recommended to have on user model.

Only fields related to `Authentication (email, password) and authorization (role, active state)` should be on User model. All others are on profile

**Review by feyton — CHANGES_REQUESTED** [review]
@jeandamascyifuzo 
This is really great and the error display is now more informative. I have left some comments that needs to be addressed.

**Review by feyton — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
@jeandamascyifuzo Nice work,
Error messages are handled. Work of fixing the conflicts.

File: src/index.ts (hunk 1 above)
Comment (rukundo-kevin) [reply src/index.ts]: > That would be the best compromise in between. We need errors for debugging our code when things break.
We can try based on the NODE_ENV

Can you try based this or write a formatError function that can log the stacktrace then return the error without the stacktrace.

**Comment by feyton** [comment]
[Custom error response](https://trello.com/c/5lqLN7aW/98-custom-error-response)

**Review by rukundo-kevin — APPROVED** [review]

**Pull request merged**